packages/@n8n/config/src/decorators.ts

```diff
@@ -46,6 +47,8 @@ export const Config: ClassDecorator = (ConfigClass: Class) => {
 					} else {
 						value = value === 'true';
 					}
+				} else if (type !== String && type !== Object) {
+					value = new (type as Constructable)(value as string);
```

Comment:
we should eventually have all this parsing code in a try-catch for better errors. but that can also be deferred for now.